waddrmgr/db.go

```diff
@@ -744,6 +749,35 @@ func fetchAddress(tx walletdb.Tx, addressID []byte) (interface{}, error) {
 	return nil, managerError(ErrDatabase, str, nil)
 }
 
+// markAddressUsed flags the provided address id as used in the database.
+func markAddressUsed(tx walletdb.Tx, addressID []byte) error {
+	bucket := tx.RootBucket().Bucket(usedAddrBucketName)
+
+	addrHash := fastsha256.Sum256(addressID)
+	val := bucket.Get(addrHash[:])
+	if val != nil {
+		return nil
+	}
+	err := bucket.Put(addrHash[:], []byte{0})
+	if err != nil {
+		str := fmt.Sprintf("failed to mark address used %x", addressID)
+		return managerError(ErrDatabase, str, err)
+	}
+	return nil
+}
+
+// fetchAddrUsed returns true if the provided address id was flagged as used.
+func fetchAddrUsed(tx walletdb.Tx, addressID []byte) bool {
+	bucket := tx.RootBucket().Bucket(usedAddrBucketName)
+
+	addrHash := fastsha256.Sum256(addressID)
```

Comment:
Since this function is only called internally from `fetchAddress` and it already has the hash, I'd suggest changing the signature to `addrHash []byte` so we can avoiding needlessly calculating the hash again.